Install.lua

```diff
@@ -1,175 +1,275 @@
+			if ScriptClass == "mod" or ScriptClass == "module" then TypesSpecified = true end
+		end
 
-MakeScript(ReplicatedStorage, "App/NevermoreEngine.lua")
+		if MainExists then Directory = Directory + 1 end -- :gsub("[^/]+$", "", 1) end			
+		local Count = 0
+
+		local function LocateFolder(FolderName)
+			Count = Count + 1
+			if Count > Directory then
+				Directory = Directory + 1
+				if (Parent and Parent.Name) ~= FolderName and "Modules" ~= FolderName then
+					if FolderName == "App" and Link:find("/Quenty/NevermoreEngine/", 1, true) then FolderName = "ReplicatedStorage" end
+					local Success, Service = pcall(game.GetService, game, FolderName)
+					if FolderName ~= "Lighting" and Success and Service then
+						Parent = Service
+					else
+						if Generated then
+							Parent = Instance.new("Folder", Parent)
+							Parent.Name = FolderName
+							if not Routines[1] then Routines[1] = Parent end
+							DataSources[Parent] = "https://github.com" .. (Sub:match(("/[^/]+"):rep(Directory > 1 and Directory + 2 or Directory)) or warn("[1]", Sub, Directory > 1 and Directory + 2 or Directory) or "")
+						else
+							local Object = Parent and Parent:FindFirstChild(FolderName)
+							if not Object then
+								Object = Instance.new("Folder", Parent)
+								Object.Name = FolderName
+								if not Routines[1] then Routines[1] = Object end
+								DataSources[Object] = "https://github.com" .. (Sub:match(("/[^/]+"):rep(Directory > 1 and Directory + 2 or Directory)) or warn("[2]", Sub, Directory > 1 and Directory + 2 or Directory) or "")
+								Generated = true
+							end
+							Parent = Object
+						end
+					end
+				end
+			end
+		end
+		
+		Link:gsub("[^/]+/?$", ""):gsub("[^/]+", LocateFolder)
+
+		if MainExists or ScriptCount ~= 1 or ScriptName ~= LastFolder then
+			LocateFolder(LastFolder)
+		end
+		
+		local Script = Instance.new(ScriptTypes[ScriptClass or TypesSpecified and "" or "mod"], Parent)
+		Script.Name = ScriptName
+		if not Routines[1] then Routines[1] = Script end
+		coroutine.resume(coroutine.create(GiveSourceToScript), "https://raw.githubusercontent.com" .. ScriptLink:gsub("(/[^/]+/[^/]+/)blob/", "%1", 1), Script)
+		
+		if MainExists then
+			Parent = Script
+		end
 
-local MainDirectory = GetDirectory(ServerScriptService, "Nevermore\\")
-local Paths = {}
-for ScriptPath in (LoadURL("Modules/ModuleList.txt")):gmatch("[^\r\n]+") do
-	Paths[#Paths+1] = ScriptPath
-	print""
+		for a = 2, ScriptCount do
+			local Link = Scripts[a]
+			local ScriptName, ScriptClass = Link:match("([%w-_%%]+)%.?(%l*)%.lua$")
+			local Script = Instance.new(ScriptTypes[ScriptClass or TypesSpecified and "" or "mod"], Parent)
+			Script.Name = ScriptName:gsub("Library$", "", 1):gsub("%%(%x%x)", UrlDecode)
+			coroutine.resume(coroutine.create(GiveSourceToScript), "https://raw.githubusercontent.com" .. Link:gsub("(/[^/]+/[^/]+/)blob/", "%1", 1), Script)
```

Comment:
coroutine passing like this is hard to follow, need to be careful in certain cases. DownloadScriptSource maybe better name? Concatenation of link here hard to follow.